gdal.cmake

```
@@ -199,6 +198,10 @@ else ()
 
 endif ()
 
+if (CMAKE_BUILD_TYPE MATCHES Debug)
+  add_definitions(-DDEBUG)
+endif ()
```

This command is superseded. I think this should this now (untested):
```suggestion
add_compile_options($<$<CONFIG:DEBUG>:-DDEBUG>)
```

---

Good suggestion. Using actually ``add_compile_definitions($<$<CONFIG:DEBUG>:DEBUG>)``, since it is a compile definition rather than a compile option

---

Ah true, I usually just add options to adjust the compiler warning levels, so I forgot there is a special command for definitions 😅 .

Btw: there are more occurrences of `add_definition` in this project, even this is maybe nothing for this PR.